lib/crono_trigger/polling_thread.rb

```diff
@@ -76,13 +76,20 @@ def poll(model)
               end
             end
           rescue Concurrent::RejectedExecutionError
-            overflowed_records << record
+            overflowed_record_ids << record.id
           end
         end
-        model.connection_pool.with_connection do
-          model.where(id: overflowed_records).crono_trigger_unlock_all!
-        end
-      end while overflowed_records.empty? && records.any?
+        unlock_overflowed_records(model, overflowed_record_ids)
+      end while overflowed_record_ids.empty? && records.any?
+    end
+
+    private def unlock_overflowed_records(model, overflowed_record_ids)
+      model.connection_pool.with_connection do
+        model.where(id: overflowed_record_ids).crono_trigger_unlock_all!
+      end
+    rescue ActiveRecord::ConnectionNotEstablished, ActiveRecord::LockWaitTimeout, ActiveRecord::StatementTimeout, ActiveRecord::Deadlocked
```

Comment:
`ActiveRecord::ConnectionTimeoutError` should be added.
We received an error like below:

> E, [2018-09-18T10:45:14.446951 #64] ERROR -- : could not obtain a connection from the pool within 5.000 seconds (waited 5.000 seconds); all pooled connections were in use (ActiveRecord::ConnectionTimeoutError)

Reply:
`ActiveRecord::ConnectionTimeoutError` is subclass of `ActiveRecord::ConnectionNotEstablished`.

Reply:
I see 🙆 

```
[1] pry(main)> ActiveRecord::ConnectionTimeoutError.ancestors
=> [ActiveRecord::ConnectionTimeoutError,
 ActiveRecord::ConnectionNotEstablished,
 ActiveRecord::ActiveRecordError,
 StandardError,
 Exception,
 ActiveSupport::Dependencies::Blamable,
 ActiveSupport::ToJsonWithActiveSupportEncoder,
 Object,
 Tins::Full,
 Tapp::ObjectExtension,
 PP::ObjectMixin,
 ActiveSupport::Tryable,
 JSON::Ext::Generator::GeneratorMethods::Object,
 ActiveSupport::Dependencies::Loadable,
 Kernel,
 BasicObject]
```